docs/examples/example_onnx.ipynb

```diff
@@ -0,0 +1,349 @@
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "from pathlib import Path\n",
+    "from git_root import git_root\n",
+    "\n",
+    "df = pd.read_csv(git_root(\"data/learning_mindset.zip\"))\n",
+    "outcome_column = \"achievement_score\"\n",
+    "treatment_column = \"intervention\"\n",
+    "feature_columns = [\n",
+    "    column for column in df.columns if column not in [outcome_column, treatment_column]\n",
+    "]\n",
+    "categorical_feature_columns = [\n",
+    "    \"ethnicity\",\n",
+    "    \"gender\",\n",
+    "    \"frst_in_family\",\n",
+    "    \"school_urbanicity\",\n",
+    "    \"schoolid\",\n",
+    "]\n",
+    "# Note that explicitly setting the dtype of these features to category\n",
+    "# allows both lightgbm as well as shap plots to\n",
+    "# 1. Operate on features which are not of type int, bool or float\n",
+    "# 2. Correctly interpret categoricals with int values to be\n",
+    "#    interpreted as categoricals, as compared to ordinals/numericals.\n",
+    "for categorical_feature_column in categorical_feature_columns:\n",
+    "    df[categorical_feature_column] = df[categorical_feature_column].astype(\"category\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now that we've loaded the experiment data, we can train a MetaLearner.\n",
+    "\n",
+    "\n",
+    "### Training a MetaLearner\n",
+    "\n",
+    "Again, mirroring our {ref}`example on estimating CATEs with a MetaLearner\n",
+    "<example-basic>`, we can train an\n",
+    "{class}`~metalearners.XLearner` as follows:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from metalearners import XLearner\n",
+    "from lightgbm import LGBMRegressor, LGBMClassifier\n",
+    "\n",
+    "xlearner = XLearner(\n",
+    "    nuisance_model_factory=LGBMRegressor,\n",
+    "    propensity_model_factory=LGBMClassifier,\n",
+    "    treatment_model_factory=LGBMRegressor,\n",
+    "    is_classification=False,\n",
+    "    n_variants=2,\n",
+    "    nuisance_model_params={\"n_estimators\": 5, \"verbose\": -1},\n",
+    "    propensity_model_params={\"n_estimators\": 5, \"verbose\": -1},\n",
+    "    treatment_model_params={\"n_estimators\": 5, \"verbose\": -1},\n",
+    "    n_folds=2,\n",
+    ")\n",
+    "\n",
+    "xlearner.fit(\n",
+    "    X=df[feature_columns],\n",
+    "    y=df[outcome_column],\n",
+    "    w=df[treatment_column],\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```{note}\n",
+    "In this example we used all ``lightgbm`` models because these are the only type of models\n",
+    "that we managed to make them work with the categorical encoding from ``pandas``\n",
+    "and their conversion to ONNX. Other ``sklearn`` models which support categoricals such as\n",
```

Comment:
```suggestion
    "In this example, we used all ``lightgbm`` models because these are the only type of models\n",
    "that we managed to get to work with categorical encodings from ``pandas``\n",
    "while also being convertible to ONNX. Other ``sklearn`` models which support categoricals such as\n",
```